service/gcs/bridge/bridge.go

```diff
@@ -301,7 +268,19 @@ func (b *Bridge) ListenAndServe(bridgeIn io.ReadCloser, bridgeOut io.WriteCloser
 					"message-id":   header.ID,
 					"message":      string(message),
 				}).Debug("opengcs::bridge - read message")
-				requestChan <- &Request{header, message, b.protVer}
+				base := prot.MessageBase{}
+				if err := json.Unmarshal(message, &base); err != nil {
+					// Set the ActivityID to empty guid since the json message
+					// is invalid
+					base.ActivityID = "00000000-0000-0000-0000-000000000000"
```

Comment:
If you mark ActivityID as omitempty, then you don't have to set it at all. And indeed, that would fix an existing bug, where if the host omits ActivityID, the guest should not send it back as an empty string in the JSON, it should omit it as well. (I hit this bug while working on the external bridge and had to remove omitempty from ActivityID on the host side. But it should be safe to add here on the guest side.)

Reply:
I thought that was a protocol failure? Are you sure you can have an empty ActivityID?

Reply:
As long as it's omitempty, it will get defaulted correctly on the host (as the empty GUID). If it's not omitempty, then you're in trouble because the host cannot unmarshal empty string as empty GUID.